api/libproject/devstart.rb

```diff
@@ -751,9 +751,21 @@ def circle_deploy(cmd_name, args)
     end
   end
 
-  version = ENV.fetch("CIRCLE_TAG", "circle-ci-test")
-  promote = ENV["CIRCLE_BRANCH"] == "master" ? "--promote" : "--no-promote"
-  deploy(cmd_name, args + %W{--version #{version} #{promote}})
+  default_version = "circle-ci-test"
+  promote = "--no-promote"
+  if ENV["CIRCLE_BRANCH"] == "master"
+    # Note that --promote will generally be a no-op, as we expect
```

Comment:
So with this change, are we still deploying non-master branches to AppEngine?

I'm not sure we should be...

Reply:
Deploying but never routing traffic. I liked @calbach's perspective that we could "learn something" if this ever fails... but will also accumulate forever without cleanup. It also leaves these services exposed forever via versioned urls, with whatever security issues they may have introduced (against our test dataset of course). Maybe for non-master deploys we could deploy-then-delete.

Reply:
I don't believe RDR does this, FWIW.

We do run up a server in Circle and hit it with some requests, but I don't think it's deployed to AE.

I'm not sure what we'd learn from deploying to AppEngine here, particularly if we're not hitting it afterwards?

Reply:
Yeah, RDR doesn't do this. The kinds of things that could fail include, say, permissions (like if we revoke a token) -- which are nice to learn about at the earliest opportunity, but I wouldn't mind if we skipped this.

Reply:
I'd be happy to learn those things after the merge. 

(Getting Circle status back in PRs faster, and not burdening AppEngine with a bunch of versions we don't use, seems more important.)

Reply:
OK - I think lets not bother pushing at all outside of master. I left it in because I'm pretty sure that was the intent with the previous behavior and so I rationalized that this was to catch early deployment issues. But if I were defining this from the scratch I probably wouldn't push here.

Will fix this and the other issue Josh noticed in about 30m when I'm at my desk.